packages/renderer/src/lib/node/NodeDetails.spec.ts

```diff
@@ -38,63 +40,136 @@ const node: V1Node = {
   },
 } as V1Node;
 
-vi.mock('/@/stores/kubernetes-contexts-state', async () => {
+vi.mock(import('../kube/resources-listen'), async importOriginal => {
+  // we want to keep the original nonVerbose
+  const original = await importOriginal();
   return {
-    kubernetesCurrentContextNodes: vi.fn(),
+    ...original,
+    listenResources: vi.fn(),
+    isKubernetesExperimentalMode: vi.fn(),
   };
 });
 
-beforeAll(() => {
-  Object.defineProperty(window, 'kubernetesReadNode', { value: vi.fn() });
+vi.mock('/@/stores/kubernetes-contexts-state');
+
+beforeEach(() => {
+  vi.resetAllMocks();
+  router.goto('http://localhost:3000');
 });
 
-test('Confirm renders node details', async () => {
-  // mock object store
-  const nodes = writable<KubernetesObject[]>([node]);
-  vi.mocked(kubeContextStore).kubernetesCurrentContextNodes = nodes;
+type initListsReturnType = {
+  updateNodes: (objects: KubernetesObject[]) => void;
+  updateEvents: (objects: CoreV1Event[]) => void;
+};
+
+describe.each<{
+  experimental: boolean;
+  initLists: (nodes: KubernetesObject[], events: CoreV1Event[]) => initListsReturnType;
+}>([
+  {
+    experimental: false,
+    initLists: (nodes: KubernetesObject[], events: CoreV1Event[]): initListsReturnType => {
```

Comment:
I will factorize this code which is duplicated in ServiceDetails and DeploymentDetails tests in a specific PR